titus-ext/cassandra/src/main/java/com/netflix/titus/ext/cassandra/store/CassandraJobStore.java

```diff
@@ -606,43 +606,62 @@ public Completable deleteTask(Task task) {
 
     @Override
     public Observable<Job<?>> retrieveArchivedJob(String jobId) {
-        return Observable.fromCallable((Callable<Statement>) () -> retrieveArchivedJobStatement.bind(jobId)).flatMap(statement -> execute(statement)
-                .map(resultSet -> {
-                    Row row = resultSet.one();
-                    if (row == null) {
-                        throw JobStoreException.jobDoesNotExist(jobId);
-                    }
-                    String value = row.getString(0);
-                    return (Job<?>) ObjectMappers.readValue(mapper, value, Job.class);
-                }));
+        Observable<Job> action = retrieveEntityById(jobId, Job.class, retrieveArchivedJobStatement)
+                .switchIfEmpty(retrieveEntityById(jobId, Job.class, retrieveActiveJobStatement))
+                .switchIfEmpty(Observable.error(JobStoreException.jobDoesNotExist(jobId)));
+        return (Observable) action;
     }
 
     @Override
     public Observable<Task> retrieveArchivedTasksForJob(String jobId) {
```

Comment:
please add some documentation here (and consider renaming the method) since now it will also always returns active tasks for the job. It seems to be handled correctly by the only current caller, but it may cause confusion in the future

Reply:
Active state should never be read like that (all of it is read on startup). But technically you are right that we could read active data with this method. I will add a comment documenting this.